src/node_http_parser.cc

```diff
@@ -636,6 +644,24 @@ class Parser : public AsyncWrap, public StreamListener {
     if (ret.IsEmpty())
       return;
 
+    // check header parsing time
+    if (header_parsing_start_time_ != 0 && headers_timeout_ != 0) {
+      auto now = uv_hrtime();
+      auto parsing_time = (now - header_parsing_start_time_) / 1e6;
+
+      if (parsing_time > headers_timeout_) {
+        Local<Value> cb =
+            object()->Get(env()->context(), kOnTimeout).ToLocalChecked();
+
+        if (!cb->IsFunction())
+          return;
+
+        MakeCallback(cb.As<Function>(), 0, nullptr);
```

Comment:
should I check the return value? anything else needed to mark the parsing error?

Reply:
@OrKoN Since you’d be `return;`ing here either way, whether the call threw an exception or not, I don’t think so. You can wrap the call in `USE()` to make it explicit that you’re discarding the return value because of that.